File: Ejemplo en C/Clases/Arma.c
#include <stdlib.h>
#include <stdio.h>
#include "Arma.h"

Arma *Crear_Arma(int municion, float tiempo_recarga, int daño)
{
    Arma *nueva_arma = (Arma *) malloc(sizeof(Arma));

    nueva_arma->municion = municion;
    nueva_arma->tiempo_recarga = tiempo_recarga;
    nueva_arma->daño = daño;

    return nueva_arma;
}

void Destruir_Arma(Arma *esta_arma){
    free(esta_arma);
}

void Disparar(Enemigo *objetivo, Arma *esta_arma) 
{
    return "OWO"; //polimorfismo?
}

char *ToString()
{
    return "Arma"; //polimorfismo?
}

File: Ejemplo_en_C#/Para_Visual_Studio/Programa_Video_Juego/Clases/Arma.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Programa_Video_Juego.Clases
{
    abstract class Arma
    {
        protected int municion;
        protected float tiempo_recarga;
        protected int daño;

        public Arma(int municion, float tiempo_recarga, int daño)
        {
            Municion = municion;
            Tiempo_recarga = tiempo_recarga;
            Daño = daño;
        }

        public int Municion { get => municion; set => municion = value; }
        public float Tiempo_recarga { get => tiempo_recarga; set => tiempo_recarga = value; }
        public int Daño { get => daño; set => daño = value; }

        public abstract string Disparar(Enemigo objetivo);
    }
}
